### ISSUE TITLE
Issue when using external file for intervals.

### ISSUE THREAD

**Issue by mutabledevotions** [satyarth/pixelsort#9]
For sake of testing this, I'm trying to sort this image:
![pc](https://user-images.githubusercontent.com/4504400/35190716-63b087f4-fe2d-11e7-84d8-bc2f709377dc.png)

Using this external interval file:
![mask](https://user-images.githubusercontent.com/4504400/35190717-6af016e2-fe2d-11e7-8338-21e4d6a6cde8.png)
 
The operation:
`λ python pixelsort.py "pclips/pclips.png" -of pclips -a 45 -f "pclips/mask.png" -i file`

Returns the following:
```
Interval function:  file
Randomness:  0 %
Opening image...
Converting to RGBA...
Rotating image...
Getting data...
Getting pixels...
Determining intervals...
Cleaning up edges...
Traceback (most recent call last):
  File "pixelsort.py", line 56, in <module>
    main()
  File "pixelsort.py", line 31, in main
    intervals = argparams.interval_function(pixels, argparams)
  File "C:\Users\justi\Code\pixelsort\interval.py", line 117, in file_mask
    if file_pixels[y][x] == constants.black_pixel and file_pixels[y][x - 1] == constants.black_pixel:
IndexError: list index out of range
```

I have also tested this on couple other bitmap mask files -- is there something I'm missing? Black pixels too close to the edge?

**Comment by BernardZhao** [comment]
I think this has to do with using the -a angle parameter while also using a file interval input. I also got the same error, but only when the angle was not 0.

**Comment by BernardZhao** [comment]
#10 Fixed this issue, this can be closed now.

**Comment by satyarth** [comment]
Thanks a lot @justinwzig  and @BernardZhao! Sorry for not being responsive.

**Comment by agusakov** [comment]
Hi, I'm also getting this issue, but I'm not doing anything with angles.
Here's the operation I used:
`pixelsort$ python3 pixelsort.py ../Downloads/Inverted_Lion-1.jpg -i file -f ../Desktop/intervals.png`

And here's the error I got, same as the one above:

> Interval function:  file
Randomness:  0 %
Opening image...
Converting to RGBA...
Rotating image...
Getting data...
Getting pixels...
Determining intervals...
Cleaning up edges...
Traceback (most recent call last):
  File "pixelsort.py", line 56, in <module>
    main()
  File "pixelsort.py", line 31, in main
    intervals = argparams.interval_function(pixels, argparams)
  File "/Users/alenagusakov/pixelsort/interval.py", line 118, in file_mask
    if file_pixels[y][x] == constants.black_pixel and file_pixels[y][x - 1] == constants.black_pixel:
IndexError: list index out of range

Edit 1: I think the issue might be the fact that the images are differently sized perhaps?
Edit 2: yup, resized the interval image, fixed the problem.
